xla/util.h

```diff
@@ -1018,6 +1018,45 @@ struct FreeDeleter {
 std::unique_ptr<void, FreeDeleter> AlignedAlloc(std::size_t alignment,
                                                 std::size_t size);
 
+// Serialized static variable initialization from the `func` output.
+// It avoid a deadlock between static variable initialization lock
```

Comment:
I'd note that:
a) this does not hold a lock while func() is called, which means that:
b) this may call func multiple times if invoked concurrently, but the output from all but one will be discarded.

It also occurs to me that it might be better to use a PEP 703 critical section here, since it would avoid the multiple call most of the time. For example in free threading mode, we could do this:

```
static nb::object* mutex_object = new nb::object();

nb::ft_object_guard lock(*mutex_object);
if (!initialized) {
  init();
}
```

I believe this will avoid the deadlock also because outer critical sections are released if you acquire an inner critical section.

Reply:
(But note you'd need to use something like an absl::Mutex in non-FT mode.)

Reply:
Actually, never mind. You wouldn't need an `absl::Mutex`: the GIL would suffice.

Reply:
We finally agreed in another chat to use absl::Mutex